subsys/net/lib/http/http_client.c

```diff
@@ -423,6 +424,17 @@ static int http_wait_data(int sock, struct http_request *req)
 			LOG_DBG("Connection closed");
 			ret = total_received;
 			break;
+		} else if (received == -EAGAIN) {
+			/* Retry */
+			if (retry > CONFIG_HTTP_CLIENT_RECV_RETRY) {
+				LOG_ERR("Receiver retry failed (%d)", errno);
+				ret = -errno;
+				break;
+			}
+			LOG_DBG("Receiver retry (%d)", retry);
+			k_msleep(CONFIG_HTTP_CLIENT_RETRY_INTERVAL_MS);
```

Comment:
This will block the application now and waits always the timeout. Could we use `poll()` so that if we receive the data, we can immediately call `recv()` instead of waiting possibly long time.